### PR TITLE
Move web content from OSCAL, and other streamlining.

### PR DESCRIPTION
**Pull request by Compton-NIST**
# Committer Notes

In support of Issue #1802 

**Currently in rapid prototype mode with the team.**

- Website content moved to OSCAL-Pages project.  This project will build and publish the public website.
- Reference content moved to OSCAL-Reference project. This project will build and publish the developer documentation. AKA OSCAL Reference.
- OSCAL repo in this branch has been reduced to only the model content.

Remaining to determine:

- Linking up the websites so that it is least disruptive to the public.
- The generated xml and json content has been removed from this branch.  We need to talk about this, and what is the "right" way moving forward.  I think there are a number of options and opinions to consider.
- Figuring out the entry point into the public website. (linking, landing page, etc)

Benefits:

- Less activity on the OSCAL project unrelated to models.
- Will make it easier for the community to monitor.
- More focused workstreams between model development, website content/education and experiments with models.
- Depending on how we publish the schemas-and-converters artifact, may make it easier for the public to access and use releases of OSCAL with less tooling.
- Reference will be in position to be updated and enhanced over time for development use.
- Public website can update at-will without impacting the OSCAL model project.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

(For reviewers: [The wiki has guidance](https://github.com/usnistgov/OSCAL/wiki/Issue-Review) on code review and overall issue review for completeness.)

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [x ] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -1,62 +0,0 @@
-name: Link Check
```
Comment (aj-stein-nist) [inline .github/workflows/link-check.yml:1]: Non-blocking question: we think link checking of Markdown content doesn't have value? Removing this file gets rid of link checks for site and Markdown in-repo docs. I am fine with that if that is the implied view of the team here based on this.

**Comment by nikitawootten-nist** [comment]
Hi Chris, as part of this PR I'd like to propose some additional removals/changes:
- `src/assets`: Layer diagrams can be retired or moved to the OSCAL Pages site?
- `src/content`: Already marked as moved to `oscal-content` repo'
- `src/metaschema/examples`: A better place for this is in the metaschema repo
- `src/utils`: Should be reorganized:
    - `util`: Contents should be absorbed into `src/utils` (no need for `src/utils/util`)
- `git-cheat-sheet.md`: Can be moved safely to the wiki
- `slides-2023-02-09.pdf`: Should be removed

File: .github/workflows/status-ci-cd.yml [hunk 2]
```diff
@@ -28,29 +26,3 @@ jobs:
       commit_resources: true
     secrets:
       access_token: ${{ secrets.COMMIT_TOKEN }}
-  validate-website-reference:
```
Comment (aj-stein-nist) [inline .github/workflows/status-ci-cd.yml:31]: Question, unclear if it is blocking: we need to determine how we will generate schemas in these pipelines even if they are not committed inline anymore, and that means adapting metaschema-xslt, in this PR branch or another.

File: .github/workflows/workflow-validate-repo-markdown.yml [hunk 3]
```diff
@@ -1,70 +0,0 @@
-name: Validate Repo Markdown
```
Comment (aj-stein-nist) [inline .github/workflows/workflow-validate-repo-markdown.yml:1]: Question, non-blocking: ditto on link checking and team perspective.

**Review by aj-stein-nist — COMMENTED** [review]
This is looking good so far. I have added a few minor comments. In addition, I am not sure deleting `xml/README.md` and `json/README.md` is aggressive enough for a recursive delete if this makes sense, let me know.

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (Compton-NIST) [reply .github/workflows/link-check.yml:1]: I added the markdown checker workflow back to the branch.  That's all we need, I think.

**Comment by iMichaela** [comment]
@aj-stein-nist  and @Compton-NIST:
The file ./versioning-and-branching.md in the https://github.com/usnistgov/OSCAL/tree/feature-1802-oscal-pages branch reads:
 
_All individual work will be done in branches **in a personal fork of this repository.** 
Personal branches should be named using the convention <issue #>-brief-dashed-name.
Once work is complete on a personal branch, the branch should be interactively rebased to tidy any commits. Then a PR should be opened against the target feature-*  branch or the develop branch if the changes are to be included in the next release._ 

During my chat with AJ today around #1798, he indicated that working on a personal fork is no longer mandated. Should this file be updated?

**Comment by wendellpiez** [comment]
If given the option I intend to continue to work in a personal fork.

However, I did not interpret the new guideline as discouraging this, only as not requiring it. However I note that work when not *individual* does not have to go in a personal fork, the alternative presumably being a feature branch in the main repo.

Maybe we only need guidance on when to create a feature branch vs work in a fork. 🌻

File: .github/workflows/workflow-validate-repo-markdown.yml (hunk 3 above)
Comment (Compton-NIST) [reply .github/workflows/workflow-validate-repo-markdown.yml]: Restored this since it was needed.

File: .github/workflows/status-ci-cd.yml (hunk 2 above)
Comment (Compton-NIST) [reply .github/workflows/status-ci-cd.yml:31]: I'm not sure if this has been addressed or not, or if it is still a potential issue -- @nikitawootten-nist ?

Comment (nikitawootten-nist) [reply .github/workflows/status-ci-cd.yml:31]: If we are planning on removing auto-comitting actions, we should probably do the following:
1. Streamline the schema and converter generation to be as simple to invoke as possible (another makefile?)
2. Modify the existing action to build the schemas and converters without committing for PRs
3. Also make the action upload schema artifacts on release

For existing releases we'll have to manually upload the correct artifacts (which is simple but a bit tedious, thankfully we only have a few releases so it shouldn't be too painful)

**Review by nikitawootten-nist — CHANGES_REQUESTED** [review]
Somehow there's been a regression and the `src/utils/resolver-2018` which I had previously deleted has reappeared.

**Review by iMichaela — APPROVED** [review]
Thank you for the fantastic work done to split the OSCAL repo into multiple content-specific repos. Kudos! I approve the PR moving the website. I have some small comments, that, if not addressed now, we can address them later when doing a fine tunning.

**Comment by Compton-NIST** [comment]
I'm going to close this and reopen with the feature branch that was not squashed so that we have a more clean rebase to the updated develop branch.

**Pull request closed without merging**